source/materials.c

```diff
@@ -93,6 +93,11 @@ int aws_cryptosdk_keyring_on_encrypt(
     struct aws_array_list *edks,
     const struct aws_hash_table *enc_ctx,
     enum aws_cryptosdk_alg_id alg) {
+    AWS_PRECONDITION(aws_cryptosdk_keyring_is_valid(keyring) && (keyring->vtable != NULL));
+    AWS_PRECONDITION(aws_byte_buf_is_valid(unencrypted_data_key));
+    AWS_PRECONDITION(aws_cryptosdk_edk_list_is_valid(edks) && aws_cryptosdk_edk_list_elements_are_valid(edks));
+    AWS_PRECONDITION(aws_hash_table_is_valid(enc_ctx));
```

Comment:
@ttjsu-aws do you confirm these pre-conditions? Do we accept `request_alloc == NULL`? Do we accept `keyring_trace == NULL`? The proof does not expose any memory-safety implication for `request_alloc == NULL` or `keyring_trace == NULL`.